---
hide:
  - navigation
---
# 

To advance our understanding of how the architecture of biological systems impacts their function, I engineer analysis methods and bioinstrumentation to quantify morphology and organization across all biologically-relevant length scales. In collaboration with biologists and theoreticians, I apply these methods to study the physical and chemical mechanisms that shape our cells, tissues, and organs.

You can find a full list of my papers and patents on <u><a href="https://scholar.google.com/citations?user=zeiZjPAAAAAJ&hl=en&oi=ao">my Google Scholar profile</a>.</u>

## Multimodal spatial omics
Morphogenesis is a carefully choreographed dance of molecular and morphological patterning. Yet, it remains difficult to quantify interactions between molecular and morphological features in biological organisms. Towards bridging this gap, I develop computational frameworks for integrating spatial molecular profiling and imaging data. This work comprises community-driven data standards, interactive data visualization, and performant software libraries.

- [SpatialData](https://spatialdata.scverse.org/en/latest/): an open and universal framework for multimodal spatial omics. See our [peer-reviewed paper](https://www.nature.com/articles/s41592-024-02212-x) for details.
- [starfish](https://spacetx-starfish.readthedocs.io/en/latest/): a scaleable pipeline for image-based transcriptomics. See our peer-reviewed [paper](https://joss.theoj.org/papers/10.21105/joss.02440) for details.
- [insta](https://github.com/czbiohub/instapipeline): a tool for using crowd-sourced annotations to calibrate and validate in situ transcriptomics image processing pipelines. See our [preprint](https://www.biorxiv.org/content/10.1101/2020.07.14.201384v3) and [peer-reviewed paper](https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1009274) for details.


## Human-in-the-loop data visualization and analysis

Owing to advances in imaging and computation, there is an opportunity to observe and quantify the architecture of organisms across all biologically-relevant length scales. However, due to the size and complexity, it remains challenging to view and explore these data. To address this need, I have developed software for interactive viewing, annotation, and analysis of complex imaging data.

- [napari](https://napari.org/):  python-based, GPU-accelerated n-dimensional image viewer for annotating images and exploring analysis results that integrates with modern deep learning algorithms and big data tools.
- [napari-threedee](https://napari-threedee.github.io/): toolkit for building interactive applications for 3D annotation and analysis of images. See our pre-print [here](https://www.biorxiv.org/content/10.1101/2023.07.28.550950v1).

## Computational biomechanics
To elucidate the principles by which chemistry and physics shape biology, I develop methods for linking biomechical models to experimental data. In collaboration with experimentalists, I apply these methods to study how mechanical forces impact the form and function of biological systems from the cellular to whole organ scales.

- At the organ scale, we showed how mechanical stresses drive the emergence of the fractal architecture of the lung during development. See our pre-print [here](https://www.biorxiv.org/content/10.1101/2025.01.13.632436v1.abstract).
- At the cellular and tissue scale, we discovered how mechanical buckling instabilities inform bladder cancer tumor morphology. See our [peer-reviewed paper](https://www.nature.com/articles/s41567-024-02735-2) and the [News and Views](https://www.nature.com/articles/s41567-024-02763-y) for details.

## Microfluidic tools for single-cell analysis
Protein localization and post translational mot this need, as a graduate student in the Herr Lab, I developed microfluidic tools for single cell protein analysis. Levdifications are key to understanding cell state, yet remain difficult to measure with single cell resolution. To meeeraging the favorable mass transport scaling of microfluidic length scales, we extended the sensitivity of traditional protein assays. In particular, we developed tools for measuring the [subcellular localization of proteins from single cells](https://www.nature.com/articles/micronano201679) and for [highly-selective measurement of protein isoforms from single cells](https://onlinelibrary.wiley.com/doi/full/10.1002/anie.201606039?casa_token=PXqKDZ5-2lYAAAAA%3Ao7Wb8OH3q7nzUN_WKHpYSQ0KO1R2JPGbISd5cXhkB7Bc4HCKGH4UBda3Kk49-QJBQtCl7bBAq7LqDHtC).
